ui/components/app/confirm-page-container/confirm-page-container-content/confirm-page-container-summary/confirm-page-container-summary.component.js

```diff
@@ -17,8 +25,18 @@ const ConfirmPageContainerSummary = (props) => {
     origin,
     hideTitle,
     image,
+    transactionType,
+    toAddress,
   } = props;
 
+  const [showNicknamePopovers, setShowNicknamePopovers] = useState(false);
+  const t = useI18nContext();
+  const { toName, isTrusted } = useAddressDetails(toAddress);
+  const isContractTypeTransaction =
+    transactionType === TRANSACTION_TYPES.DEPLOY_CONTRACT ||
+    transactionType === TRANSACTION_TYPES.CONTRACT_INTERACTION;
```

Comment:
Should we consider including the contract address for known contract methods like `transferFrom`, `transfer`, `approve`?
Here's an NFT `transferFrom` call for instance
<img width="469" alt="Screen Shot 2022-02-23 at 10 46 54 AM" src="https://user-images.githubusercontent.com/34557516/155376948-6d06e415-7225-4e92-b531-49d734d42517.png">

including the address in the summary component could be a reasonable temporary resolution for the issue @Gudahtt has raised and @SayaGT  and I documented here: https://www.notion.so/As-a-user-I-want-to-properly-identify-the-asset-that-I-m-giving-allowance-or-transferring-75ad406185e448f6bbb594083532d985

Reply:
Will review once we know this.

Reply:
@adonesky1 : we have added the address as measure of security where methodName in title is derived from data that user sends. This is to ensure that user is alerted in case data has a misleading methodName.

Reply:
Totally, I understand that, I'm just wondering aloud whether it might not be a good security feature to add it to other contract interactions as well. cc @Gudahtt @danjm @SayaGT

Reply:
Also, I am not sure if `transferFrom, transfer, approve` are contract interactions. 

From docs a contract interaction is:
```
 * @property {'contractInteraction'} CONTRACT_INTERACTION - A transaction that is
 *  interacting with a smart contract's methods that we have not treated as a special
 *  case, such as approve, transfer, and transferfrom
 ```

Reply:
Yes that is the way we are defining this particular enum value within the context of MetaMask but my point is just that `approve, transfer, and transferfrom` are all contract method calls our users regularly make and we are not providing adequate transparency about the contract in our current implementation. I think the change you're making in this PR could be a reasonable solution for providing better transparency for `approve, transfer, and transferfrom` calls as well, atleast until we have new designs for the confirmation screens. @Gudahtt @danjm thoughts on this?